instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/Instrumenter.java

```diff
@@ -232,10 +232,11 @@ public void end(
 
     if (!requestListeners.isEmpty() || !requestMetricListeners.isEmpty()) {
       long endNanos = getNanos(endTime);
-      for (RequestListener requestListener : requestListeners) {
+      // TODO (trask) call end in the reverse order that start was called?
+      for (RequestListener requestListener : requestMetricListeners) {
         requestListener.end(context, attributes, endNanos);
       }
-      for (RequestListener requestListener : requestMetricListeners) {
+      for (RequestListener requestListener : requestListeners) {
```

Comment:
Nit: do we actually need two lists of listeners? Can't we merge `requestMetricListeners` into `requestListeners`?

Reply:
I'll send that in a separate PR b/c it moves the requestListeners to after the span is started (which I think is ok, but I'd like to get eyes specifically on that change)